File: week1/omerSefaYilmaz/groupAnagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */

var groupAnagrams = function (strs) {
  const anagramMap = {};

  for (let word of strs) {
    const count = new Array(26).fill(0);

    for (let char of word) {
      count[char.charCodeAt(0) - "a".charCodeAt(0)]++;
    }

    const key = count.join("#");

    if (!anagramMap[key]) {
      anagramMap[key] = [];
    }
    anagramMap[key].push(word);
  }

  return Object.values(anagramMap);
};

File: week1/yusuffenes/groupAnagrams.py
def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
    stDict=dict()
    for i in strs:
        sortedW = "".join(sorted(i))
        if  sortedW not in stDict.keys():
            stDict[sortedW] = [i]
        else:
            stDict[sortedW].append(i)           
    print(sortedW)
    return list(stDict.values())
    